src/scripts/generate-binary-ops.py

```diff
@@ -0,0 +1,286 @@
+    # are also can be simulated with this method
+    def bitshiftright(self, op1, op2):
+        op2 = random.randint(_constants.bitmin, _constants.bitmax)
+        sum = int(hex((op1 >> ((op2 % (_constants.bitmax_exposant + 1)) & 0x1f))), 16)
+        return {'oper': '>>', 'op2': op2, 'sum': sum}
+
+    # Unsigned right bitshifting
+    def bitshiftrightzero(self, op1, op2):
+        op2 = random.randint(_constants.bitmin, _constants.bitmax)
+        if (op1 >= 0): # For positive operand 1 it works the same as bitshiftright
+            sum = int(hex((op1 >> ((op2 % (_constants.bitmax_exposant + 1)) & 0x1f))), 16)
+        else: # For negative operand 1 it needs conversion to unsigned
+            sum = (unsigned32(op1) >> (op2 % (_constants.bitmax_exposant + 1)))
+            if (sum > _constants.bitmax):
+                op2 = 1  # This way it guarantees that the summa does not go above maximum
+                sum = (unsigned32(op1) >> (op2 % (_constants.bitmax_exposant + 1)))
+        return {'oper': '>>>', 'op2': op2, 'sum': sum}
+
+    # Bitwise and
+    def bitand(self, op1, op2):
+        sum = op1 & op2
+        return {'oper': '&', 'op2': op2, 'sum': sum}
+
+    # Bitwise or
+    def bitor(self, op1, op2):
+        sum = op1 | op2
+        return {'oper': '|', 'op2': op2, 'sum': sum}
+
+    # Bitwise xor (exclusive or)
+    def bitxor(self, op1, op2):
+        sum = op2 ^ op1
+        return {'oper': '^', 'op2': op2, 'sum': sum}
+
+# Generator function
+def generate_binary_tests(options):
+    if not os.path.isdir(options.output):
+        os.makedirs(options.output)
+    SEED = options.seed
+    random.seed(SEED)
+    generated_test_counter = 0 # How many tests are written (defending against script error: null)
+    file_count = 1 # How many files are created / written
+    new_file_begin = True # If new file is opened / created
+    bitcalc = False # If we need to do bitwise calculations
+    arithcalc = False # If we need to do arithmetic calculations
+    bit = False # Actually are we doing bitwise or arithmetic calculations
+    for op in options.op:
+        # If arithmetic operators are currently needed
+        if ((op == "add") or (op == "sub") or (op == "mul") or (op == "div") or
+        (op == "mod")):
+            arithcalc = True
+
+        # If bitwise operators are currently needed
+        elif ((op == "xor") or (op == "and") or (op == "or") or (op == "shiftright")
+        or (op == "shiftleft") or (op == "shiftrightzero")):
+            bitcalc = True
+
+    file_to_open = os.path.join(options.output, _GENERATED_FILENAME.format(NUMBER = file_count))
+    for i in range(options.test_count):
+        generated_test_counter += 1
+        if (generated_test_counter % _constants.test_case_in_a_file == 0): # If a new file is needed
+            new_file_begin = True
+            file_count += 1
+            file_to_open = os.path.join(options.output, _GENERATED_FILENAME.format(NUMBER = file_count))
```

Comment:
How about the following concept: (#3 is might related as well)

Each {unary_binary} generator script **ONLY** generates test cases, and passes it's output to a specific _file writer_ script? (which cuts, formats, separates, ... the input according to what is needed.)